api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/StartingState.java

```diff
@@ -85,11 +56,54 @@ private void registerSessionAndStartStreaming() {
 
         try {
             getContext().registerSession();
+            if (!areStreamingSlotsAvailable(
+                    getZk().listSessions(), getSessionId())) {
+                return;
+            }
         } catch (Exception ex) {
             switchState(new CleanupState(ex));
             return;
         }
+
         switchState(new StreamingState());
     }
 
+    private boolean areStreamingSlotsAvailable(final Collection<Session> sessions,
+                                               final String excludeSessionId) {
```

Comment:
Would not it be better if the sessions list will be filtered before the method?
For example, the calls will be like this:
```
1. areStreamingSlotsAvailable(getZk().listSessions())
2. areStreamingSlotsAvailable(getZk().listSessions().stream().filter(v -> !v.getId().equals(getSessionId()))
```
It will make the method more straightforward and bug-prone